lib/modules/dosomething/dosomething_helpers/dosomething_helpers.module

```diff
@@ -696,3 +696,41 @@ function dosomething_helpers_get_campaign_status($start_date, $end_date) {
     return 'closed';
   }
 }
+
+/**
+ * Gets the current run for a given campaign nid and user.
+ *
+ * @param int $nid
+ * @param object $user
+ *  Optional. Will default to global $user if not supplied
+ *
+ * @return node $run
+ *  Returns the campaigns node in the correct language.
+*/
+function dosomething_helpers_get_campaign_run($nid, $user = NULL) {
+  // Setup variables
+  $node = node_load($nid);
+  if (!isset($user)) {
+    global $user;
+  }
+
+  // First check if we have a run in the user lang
+  $user_language = dosomething_global_get_language($user, $node);
+  $run_nid = dosomething_helpers_extract_field_data($node->field_current_run, $user_language);
+
+  // If not try the nodes source language
+  if (!isset($run)) {
+    $node_language = $node->language;
+    $run_nid = dosomething_helpers_extract_field_data($node->field_current_run, $node_language);
+
+    // Reutrn NULL if the campaign doesnt have a run nid
+    if (!isset($run_nid)) {
+      return NULL;
```

Comment:
It's better to return `FALSE` to be consistent with `node_load()` (as you return node_load result below).